assets/js/googlesitekit/datastore/user/date-range.js

```diff
@@ -201,12 +205,15 @@ export const selectors = {
 	 * Returns the current reference date, typically today.
 	 *
 	 * @since 1.22.0
+	 * @since n.e.x.t Added options to allow getting the reference date as a Date instance.
 	 *
-	 * @param {Object} state The current data store's state.
+	 * @param {Object} state            The current data store's state.
+	 * @param {Object} [options]        Options parameter. Default is: {}.
+	 * @param {number} [options.parsed] Number of days to offset. Default is: 0.
 	 * @return {string} The current reference date as YYYY-MM-DD.
 	 */
-	getReferenceDate( state ) {
-		return state.referenceDate;
+	getReferenceDate( state, { parsed = false } = {} ) {
+		return parsed ? new Date( state.referenceDate ) : state.referenceDate;
```

Comment:
- The description for `parsed` is inaccurate. Looks like this was copied from `offsetDays`
- I don't think we should return `Date` instances here as it can reduce the performance of `useSelect` which optimizes for shallow equality
- The return value of this selector still only documents a `string` return

Reply:
I'll create a follow up PR that fixes it.

Reply:
Actually, if we return a string date here, then it kills all sense of this eslint rule because we will need to parse the date in components using `new Date( referenceDate )`. 🤔

Reply:
Ok, i think i figured out how to do it.